collections/src/main/scala/strawman/collection/IndexedSeq.scala

```diff
@@ -27,19 +27,26 @@ trait IndexedSeqOps[+A, +CC[X] <: IndexedSeq[X], +C] extends Any with SeqOps[A,
       } else Iterator.empty.next()
   }
 
-  override def view: IndexedView[A] = new IndexedView[A] {
-    def length: Int = self.length
-    def apply(i: Int): A = self(i)
-  }
+  override def view: IndexedView[A] = new IndexedView.Id[A](this)
+
+  override protected[this] def reversed: Iterable[A] = new IndexedView.Reverse(this)
+
+  // Override transformation operations to use more efficient views than the default ones
+  override def prepended[B >: A](elem: B): CC[B] = iterableFactory.from(new IndexedView.Prepended(elem, this))
+
+  override def take(n: Int): C = fromSpecificIterable(new IndexedView.Take(this, n))
+
+  override def takeRight(n: Int): C = fromSpecificIterable(new IndexedView.TakeRight(this, n))
+
+  override def drop(n: Int): C = fromSpecificIterable(new IndexedView.Drop(this, n))
+
+  override def dropRight(n: Int): C = fromSpecificIterable(new IndexedView.DropRight(this, n))
 
-  override protected[this] def reversed: Iterable[A] = view.reverse
+  override def map[B](f: A => B): CC[B] = iterableFactory.from(new IndexedView.Map(this, f))
 
-  /** A collection containing the last `n` elements of this collection. */
-  override def takeRight(n: Int): C = fromSpecificIterable(view.takeRight(n))
+  override def reverse: C = fromSpecificIterable(new IndexedView.Reverse(this))
```

Comment:
The difference between `view.takeRight` and `IndexedView.TakeRight` (line 39) is that the former builds a view of a view, whereas the latter builds just one view. The idea is to remove one level of indirection. If that would have been optimized anyway by the JIT then we should switch to `view.takeRight` instead.